File: python/get_pull_requests.py
import requests

def get_pull_requests(github_base_url :str, user_name :str, repository :dict, headers :dict) -> tuple[dict, str]:
    github_pull_requests_url = f"{github_base_url}/repos/{user_name}/{repository['name']}/pulls"
    response = requests.request("GET", github_pull_requests_url, headers=headers)

    if response.status_code == 200:
        pull_requests = response.json()
        return pull_requests, ""
    else:
        return {}, f"Failed to retrieve pull requests. Status code: {response.status_code}"

File: go/get_pull_requests.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
)

func get_pull_requests(client *http.Client, Bearer string, repository Repository) []PullRequest {

	fetch_pull_request_url := fmt.Sprintf("%s/repos/%s/%s/pulls", GITHUB_BASE_URL, USER_NAME, repository.Name)
	req, err := http.NewRequest("GET", fetch_pull_request_url, nil)
	if err != nil {
		log.Fatalln(err)
	}
	req.Header.Set("Authorization", Bearer)

	resp, err := client.Do(req)
	if err != nil {
		log.Println("Error on response.\n[ERROR] -", err)
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		log.Println("Error while reading the response bytes:", err)
	}

	var pull_requests []PullRequest

	err = json.Unmarshal(body, &pull_requests)
	if err != nil {
		log.Println("Error while marshalling the response bytes:", err)
	}

	return pull_requests
}
